docs/source/fetching/persisted-queries.mdx

```diff
@@ -0,0 +1,160 @@
+
+1. In the `output` section of your configuration, set the [`operationManifest`](./../code-generation/codegen-configuration#operation-manifest) with:
+- a `path` to output the manifest file
+- the `version`: either `persistedQueries` for persisted queries or `legacyAPQs` for APQs
+
+```json title="apollo-codegen-config.json"
+"output": {
+    "operationManifest" : {
+        "path" : "./generated/operationIdentifiers.json",
+        "version" : "persistedQueries"
+    }
+}
+```
+
+2. In the `options` section of your configuration, set the `operationDocumentFormat` array to `definition`, `operationId`, or both `definition` and `operationId`. This determines how to generate the operation documents for your generated operations.
+
+```json title="apollo-codegen-config.json"
+"options": {
+    "operationDocumentFormat" : [
+        "definition",
+        "operationId"
+    ]
+}
+```
+
+Once you've configured code generation, you generate an operation manifest in one of two ways:
+- using the standard [`generate`](./../code-generation/codegen-cli#generate) command 
+- using the [`generate-operation-manifest`](./../code-generation/codegen-cli#generate-operation-manifest) command; this generates the operation manifest without running full code generation
+
+The resulting operation manifest for `persistedQueries` looks something like this: 
+
+```json title="operationIdentifiers.json"
+{
+  "format": "apollo-persisted-query-manifest",
+  "version": 1,
+  "operations": [
+    {
+      "id": "e0321f6b438bb42c022f633d38c19549dea9a2d55c908f64c5c6cb8403442fef",
+      "body": "query GetItem { thing { __typename } }",
+      "name": "GetItem",
+      "type": "query"
+    }
+  ]
+}
+```
+
+The resulting operation manifest JSON for `legacyAPQ` looks something like this: 
+
+```json title="operationIdentifiers.json"
+{
+  "525cd90917f8945d8d6a68fde991aaf766ff7d7b227ab2cd8dc97b7959c8d0d6" : {
+    "name": "GetItem",
+    "source": "query GetItem { thing { __typename ...Name } } fragment Name on Item { __typename name }"
+  }
+}
+```
+
+To automatically update the manifest for each new app release, you can include the [`generate`](./../code-generation/codegen-cli#generate) or [`generate-operation-manifest`](./../code-generation/codegen-cli#generate-operation-manifest) command in your CI/CD pipeline.
+
+#### Publish manifests to the PQL
+
+<blockquote>
+
+**This step is only required for persisted queries, not APQ.**
+
+Ensure your Rover CLI version is `0.17.2` or later. Previous versions of Rover don't support publishing operations to a PQL.
+[Download the latest version.](/rover/getting-started/)
+
+</blockquote>
+
+If you want to preregister your manifest as a safelist of trusted operations, you need to publish it to a PQL using the [Rover CLI](/rover/):
```

Comment:
Any interest in doing that with a Gradle task to avoid downloading an extra tool? Should be relatively easy to do if we want to.

Reply:
@AnthonyMDev @BobaFetters , thoughts on this?

Reply:
Kotlin can do this with a Gradle task, but the iOS build system does not have Gradle or an equivalent tool that is easily available. Using the Rover CLI is the best approach for now on iOS.